### PR TITLE
arch/sim: Make the SIGUSR1 host signal use the NuttX irq logic

### PR DESCRIPTION
**Pull request by sebastianene07**
## Summary

This patch updates the SIGUSR1 handling to use the NuttX interrupt logic. Remove unused sections from the sim/arch.h

## Impact

None.

## Testing

This has been verified on `sim/ostest`.

Signed-off-by: Sebastian Ene <sene@apache.org>

### CODE REVIEW

File: arch/sim/src/sim/up_internal.h [hunk 1]
```diff
@@ -242,6 +242,7 @@ void sim_cpu0_start(void);
 void up_cpu_started(void);
 int up_cpu_paused(int cpu);
 struct tcb_s *up_this_task(void);
+int host_setcpupause_request(int irq);
```
Comment (xiaoxiang781216) [inline arch/sim/src/sim/up_internal.h:245]: change to up_cpu_set_pause_handler?

**Review by xiaoxiang781216 — CHANGES_REQUESTED** [review]

File: arch/sim/src/sim/up_simsmp.c [hunk 2]
```diff
@@ -93,6 +93,8 @@ void sched_note_cpu_pause(struct tcb_s *tcb, int cpu);
 void sched_note_cpu_resume(struct tcb_s *tcb, int cpu);
 #endif
 
+void up_enable_irq(int irq);
```
[the remark this answers is not in the document]
Comment (sebastianene07) [reply arch/sim/src/sim/up_simsmp.c:96]: Yes we don't need this, I will remove it

File: arch/sim/src/sim/up_internal.h (hunk 1 above)
Comment (sebastianene07) [reply arch/sim/src/sim/up_internal.h:245]: I updated the name to `up_cpu_set_pause_handler `

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Pull request merged**